app/views/schools/register_ect_wizard/_use_previous_ect_choices.html.erb

```diff
@@ -25,19 +25,19 @@
       summary_list.with_row do |row|
         row.with_key(text: 'Delivery partner')
         row.with_value(text: @ect.previous_delivery_partner_name)
-      end 
+      end
     else
       summary_list.with_row do |row|
         row.with_key(text: 'Lead provider')
-        row.with_value(text: @ect.previous_eoi_lead_provider_name)
-      end 
-    end 
-  end 
+        row.with_value(text: @ect.previous_eoi_lead_provider_name || @school.last_chosen_lead_provider&.name)
+      end
+    end
+  end
 end %>
 
 <% if @school.provider_led_training_programme_chosen? && !@ect.lead_provider_has_confirmed_partnership_for_contract_period?(@school) %>
   <p class="govuk-body">
-    <%= @ect.previous_eoi_lead_provider_name %> will confirm if they’ll be working with your school and which delivery partner will deliver training events.
+    <%= @ect.previous_eoi_lead_provider_name || @school.last_chosen_lead_provider&.name %> will confirm if they’ll be working with your school and which delivery partner will deliver training events.
```

Comment:
Do you think we should blow up here too instead of safely rendering nothing?

```suggestion
    <%= @ect.previous_eoi_lead_provider_name || @school.last_chosen_lead_provider.name %> will confirm if they’ll be working with your school and which delivery partner will deliver training events.
```

Reply:
I'm just assuming that's why you removed it in `app/views/schools/register_mentor_wizard/review_mentor_eligibility.html.erb`